src/SFML/Window/Unix/WindowImplX11.cpp

```diff
@@ -814,19 +799,14 @@ WindowHandle WindowImplX11::getSystemHandle() const
 ////////////////////////////////////////////////////////////
 void WindowImplX11::processEvents()
 {
-    using namespace WindowsImplX11Impl;
-
     XEvent event;
 
-    // Pick out the events that are interesting for this window
-    while (XCheckIfEvent(m_display, &event, &checkEvent, reinterpret_cast<XPointer>(m_window)))
-        m_events.push_back(event);
-
-    // Handle the events for this window that we just picked out
-    while (!m_events.empty())
+    // Handle pending events
+    // Input methods might drop/change the events in the queue while they are filtered
+    // so we must process them one by one
+    while (XPending(m_display))
     {
-        event = m_events.front();
-        m_events.pop_front();
+        XNextEvent(m_display, &event);
```

Comment:
Wait a minute... XNextEvent pulls the next event from any window associated with this process... We should use XCheckWindowEvent?

Reply:
No... The IME requires ClientMessage events to be fed through XFilterEvent(), which cannot be obtained by XCheckWindowEvent.

We should fall back to using XCheckIfEvent?